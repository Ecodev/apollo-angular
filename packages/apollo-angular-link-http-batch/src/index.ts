export {HttpBatchLink, HttpBatchLinkHandler} from './http-batch-link';
export {HttpBatchLinkModule, PROVIDERS} from './http-batch-link-module';
export {BatchOptions} from './types';
