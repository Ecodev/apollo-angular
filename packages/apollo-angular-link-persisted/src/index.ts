import {DocumentNode} from 'graphql';
import {ApolloLink} from 'apollo-link';
import {setContext} from 'apollo-link-context';
import {
  createPersistedQueryLink as _createPersistedQueryLink,
  ErrorResponse,
} from 'apollo-link-persisted-queries';

export interface Options {
  generateHash?: ((document: DocumentNode) => string) | undefined;
  disable?: ((error: ErrorResponse) => boolean) | undefined;
}

const transformLink = setContext((_, context) => {
  if (context.http) {
    return {
      includeQuery: context.http.includeQuery,
      includeExtensions: context.http.includeExtensions,
    };
  }
});

export const createPersistedQueryLink = (options?: Options) =>
  // XXX: `as any` because the original package has different version of ApolloLink
  ApolloLink.from([_createPersistedQueryLink(options), transformLink as any]);
