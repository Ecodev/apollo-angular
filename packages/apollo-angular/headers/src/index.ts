import { HttpHeaders } from '@angular/common/http';
import { ApolloLink, NextLink, Operation } from '@apollo/client/core/index.js';

export const httpHeaders = () => {
  return new ApolloLink((operation: Operation, forward: NextLink) => {
    const { getContext, setContext } = operation;
    const context = getContext();

    if (context.headers) {
      setContext({
        ...context,
        headers: new HttpHeaders(context.headers),
      });
    }

    return forward(operation);
  });
};
