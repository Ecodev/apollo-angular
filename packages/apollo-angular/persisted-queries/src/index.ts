import {ApolloLink} from '@apollo/client/link/core';
import {setContext} from '@apollo/client/link/context';
import {createPersistedQueryLink as _createPersistedQueryLink} from '@apollo/client/link/persisted-queries';

export type Options = Parameters<typeof _createPersistedQueryLink>[0];

const transformLink = setContext((_, context) => {
  const ctx: any = {};

  if (context.http) {
    ctx.includeQuery = context.http.includeQuery;
    ctx.includeExtensions = context.http.includeExtensions;
  }

  if (context.fetchOptions && context.fetchOptions.method) {
    ctx.method = context.fetchOptions.method;
  }

  return ctx;
});

export const createPersistedQueryLink = (options?: Options) =>
  ApolloLink.from([_createPersistedQueryLink(options), transformLink as any]);
