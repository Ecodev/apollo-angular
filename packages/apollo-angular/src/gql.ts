import {gql as gqlTag, TypedDocumentNode} from '@apollo/client/core';

function typedGQLTag<Result, Variables>(
  literals: ReadonlyArray<string> | Readonly<string>,
  ...placeholders: any[]
): TypedDocumentNode<Result, Variables> {
  return gqlTag(literals, ...placeholders);
}

export const gql = typedGQLTag;
export const graphql = typedGQLTag;
