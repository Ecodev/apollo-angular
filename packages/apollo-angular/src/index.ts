export {Apollo, ApolloBase} from './Apollo';
export {QueryRef} from './QueryRef';
export {Query} from './Query';
export {SelectPipe} from './SelectPipe';
export {ApolloModule} from './ApolloModule';
export {APOLLO_OPTIONS} from './tokens';
