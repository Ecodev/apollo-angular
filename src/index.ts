export { SelectPipe } from './SelectPipe';
export { Apollo } from './Apollo';
export { ApolloQueryObservable } from './ApolloQueryObservable';
export { ApolloModule, APOLLO_PROVIDERS, APOLLO_DIRECTIVES, defaultApolloClient } from './ApolloModule';
