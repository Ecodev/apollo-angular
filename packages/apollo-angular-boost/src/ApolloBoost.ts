import {Injectable, Inject, Optional} from '@angular/core';
import {ApolloLink, Observable} from 'apollo-link';
import {Apollo} from 'apollo-angular';
import {HttpLink} from 'apollo-angular-link-http';
import {withClientState} from 'apollo-link-state';
import {onError} from 'apollo-link-error';
import {InMemoryCache} from 'apollo-cache-inmemory';

import {PresetConfig} from './types';
import {APOLLO_BOOST_CONFIG} from './tokens';

@Injectable()
export class ApolloBoost {
  constructor(
    private apollo: Apollo,
    private httpLink: HttpLink,
    @Optional()
    @Inject(APOLLO_BOOST_CONFIG)
    config?: PresetConfig,
  ) {
    if (config) {
      this.create(config);
    }
  }

  public create(config: PresetConfig) {
    const cache =
      config && config.cacheRedirects
        ? new InMemoryCache({cacheRedirects: config.cacheRedirects})
        : new InMemoryCache();

    const stateLink =
      config && config.clientState
        ? withClientState({...config.clientState, cache})
        : false;

    const errorLink =
      config && config.onError
        ? onError(config.onError)
        : onError(({graphQLErrors, networkError}) => {
            if (graphQLErrors) {
              graphQLErrors.map(({message, locations, path}) =>
                // tslint:disable-next-line
                console.log(
                  `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`,
                ),
              );
            }
            if (networkError) {
              // tslint:disable-next-line
              console.log(`[Network error]: ${networkError}`);
            }
          });

    const requestHandler =
      config && config.request
        ? new ApolloLink(
            (operation, forward) =>
              new Observable(observer => {
                let handle: any;
                Promise.resolve(operation)
                  .then(oper => config.request(oper))
                  .then(() => {
                    handle = forward(operation).subscribe({
                      next: observer.next.bind(observer),
                      error: observer.error.bind(observer),
                      complete: observer.complete.bind(observer),
                    });
                  })
                  .catch(observer.error.bind(observer));

                return () => {
                  if (handle) {
                    handle.unsubscribe();
                  }
                };
              }),
          )
        : false;

    const httpLink = this.httpLink.create({
      ...((config && config.httpOptions) || {}),
      uri: (config && config.uri) || '/graphql',
    });

    const link = ApolloLink.from([
      errorLink,
      requestHandler,
      stateLink,
      httpLink,
    ].filter(x => !!x) as ApolloLink[]);

    this.apollo.create({
      link,
      cache,
    });
  }
}
