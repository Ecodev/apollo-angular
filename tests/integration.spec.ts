import './_common';

import { TestBed, async } from '@angular/core/testing';
import { NgModule, Component, destroyPlatform, getPlatform } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { ServerModule, renderModule } from '@angular/platform-server';
import { ApolloClient } from 'apollo-client';

import gql from 'graphql-tag';

import 'rxjs/add/operator/take';

import { ApolloModule, Apollo } from '../src';
import { mockClient } from './_mocks';

// Mock GraphQL endpoint
const query = gql`
  query websiteInfo {
    website {
      status
    }
  }
`;
const data = {
  website: {
    status: 'online',
  },
};
const client = mockClient({
  request: { query },
  result: { data },
  delay: 500,
});
function provideClient(): ApolloClient {
  return client;
}

@Component({
  selector: 'app',
  template: 'Website: {{text}}',
})
class AsyncServerApp {
  public text = '';

  constructor(
    private apollo: Apollo,
  ) {}

  public ngOnInit() {
    this.apollo.query<any>({ query })
      .take(1)
      .subscribe(result => {
        this.text = result.data.website.status;
      });
  }
}

@NgModule({
  declarations: [AsyncServerApp],
  imports: [
    BrowserModule.withServerTransition({appId: 'async-server'}),
    ServerModule,
    ApolloModule.withClient(provideClient),
  ],
  bootstrap: [AsyncServerApp],
})
class AsyncServerModule {}

describe('integration', () => {
  beforeEach(() => {
    if (getPlatform()) {
      destroyPlatform();
    }

    TestBed.resetTestEnvironment();
  });

  describe('render', () => {
    let doc: string;
    let called: boolean;

    beforeEach(() => {
      doc = '<html><head></head><body><app></app></body></html>';
      called = false;
    });

    afterEach(() => { expect(called).toBe(true); });

    it('using renderModule should work', async(() => {
      renderModule(AsyncServerModule, { document: doc }).then(output => {
        expect(clearNgVersion(output)).toMatchSnapshot();
        called = true;
      });
    }));
  });
});

function clearNgVersion(html: string): string {
  return html.replace(/ng-version=\"[^"]+\"/, '');
}
