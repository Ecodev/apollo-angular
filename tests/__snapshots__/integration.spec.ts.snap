// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration render using renderModule should work 1`] = `"<html><head></head><body><app >Website: online</app></body></html>"`;
